### ISSUE TITLE
[React] Lazy mounting overlay components with Portal and CSS animations

### ISSUE THREAD

**Issue by fseglard** [chakra-ui/ark#1619]
### Description

Hi!

As an example, let's consider having a pretty huge form with lots of `Select` components. The `Select.Content` component is rendered using a `Portal`, with `lazyMount` and `unmountOnExit` props, and also have CSS animations on open/closed state.

Currently, because the `Portal` component is always rendered, it bloats the DOM and has a pretty huge impact in term of performances, especially on the first render.

**Is there a way to lazy mount the `Portal` component directly without loosing the CSS animations?**

If I conditionally render the `Portal` component, it works, but I loose the CSS animations, which is not a valid workaround.

As an inspiration, if we take a look at the `Dialog` component of [HeadlessUI](https://headlessui.com/react/listbox), we can see that the component is rendered using a React Portal, is lazy mounted and still have CSS animations.

### Link to Reproduction

https://codesandbox.io/s/damp-paper-sym44g?file=/src/App.js

### Steps to reproduce

1. Open your devtools in the preview window
2. Inspect the DOM
3. Check that all the `Portal` components are always rendered in the DOM even if we have enabled the lazy mount property

### Ark UI Version

1.0.0-beta.4

### Framework

- [X] React
- [ ] Solid
- [ ] Vue

### Browser

Google Chrome 118

### Additional Information

I use the `Select` component as an example, but this behavior applies to all components that render an overlay: `Combobox`, `Dialog`, `Menu`, etc.

**Comment by cschroeter** [comment]
Thanks for taking your time to report this issue. Is there a reason to use the `Portal` component? You can achieve the same results by not using them.

**Comment by fseglard** [comment]
@cschroeter How can I render the content of the components in a portal without the `Portal` component? :thinking: 

I perhaps miss your point :smile:

**Comment by cschroeter** [comment]
> Currently, because the Portal component is always rendered, it bloats the DOM and has a pretty huge impact in term of performances, especially on the first render.

So why do you use a `Portal` ? Diclosure components don't have to be rendered in a `Portal` just remove the `Portal` from your example and you get the same result without the DOM bloated.

**Comment by fseglard** [comment]
If I remove the `Portal` component, the content will be rendered at the same level as the trigger component which I don't want in most cases because it can have issues with stacking context. For example, if the parent container has an `overflow: hidden`, I need to render the content in a React Portal or it will be cropped.

Also, even I remove the `Portal` component, the DOM will still be bloated by the `Positioner` components as they're always rendered too. The `lazyMount` property can only be set on the `Content` components.

If you take a look at other UI library, for example MUI : https://mui.com/material-ui/react-menu/#basic-menu
You can see that at the initial render, the content of the overlay is not rendered (non existing in the DOM).

**Comment by cschroeter** [comment]
@fseglard 

Got it! Thanks for your feedback. @segunadebayo and I come up with a solution that should work.

```jsx
export const Basic = () => (
  <Popover.Root lazyMount unmountOnExit>
    <Popover.Trigger>
      Click Me <Popover.Indicator>Indicator</Popover.Indicator>
    </Popover.Trigger>
    <Popover.Positioner>
      <Popover.Content>
        <Popover.Title>Title</Popover.Title>
        <Popover.Description>Description</Popover.Description>
      </Popover.Content>
    </Popover.Positioner>
  </Popover.Root>
)
```

And for item based disclosure components like `Accordion` or `Tabs` it would look something like this:

```jsx
export const Basic = () => {
  const items = ['panel-1', 'panel-2', 'panel-3']
  return (
    <Accordion.Root>
      {items.map((item, id) => (
        <Accordion.Item key={id} value={item} lazyMount unmountOnExit>
          <Accordion.ItemTrigger>{item} Trigger</Accordion.ItemTrigger>
          <Accordion.ItemIndicator>Icon</Accordion.ItemIndicator>
          <Accordion.ItemContent>{item} content</Accordion.ItemContent>
        </Accordion.Item>
      ))}
    </Accordion.Root>
  )
}

This will eventually not render any dom node

**Comment by fseglard** [comment]
That would be awesome! :heart: 

And it will still be possible to use the `Portal` component in that configuration?

**Comment by cschroeter** [comment]
This will also work. A fix for this will be in the next release

**Comment by fseglard** [comment]
You guys are awesome :heart: Thanks!

**Comment by cschroeter** [comment]
This is available in 1.0 and later.
